tests/testthat/test_Plot_synergy.R

```diff
@@ -0,0 +1,214 @@
+test_that("color_ranges_from_green_to_white_to_red", {
+  expected_colors <-
+    c("#00FF00", "#3FFF3F", "#7FFF7F", "#BFFFBF", # 00ff00 is green
+      "#FFFFFF", "#FFBFBF", "#FF7F7F", "#FF3F3F", # ffffff is white
+      "#FF0000")                                  # ff0000 is red
+  expect_equal(expected_colors, .GetColors(-8,8))
+})
+
+test_that("slice_at_least_first_works", {
+  expect_equal(2:5, .SliceAtLeastFirst(c(-1,5)))  # pathological case
+  expect_equal(2:5, .SliceAtLeastFirst(c(0,5)))   # degenerate case
+  expect_equal(2:5, .SliceAtLeastFirst(c(1,5)))
+  expect_equal(2:5, .SliceAtLeastFirst(c(2,5)))
+  expect_equal(3:5, .SliceAtLeastFirst(c(3,5)))
+  expect_equal(3:5, .SliceAtLeastFirst(c(3,5,10)))  # spurious val 10
+})
+
+test_that("numeric_apply_range_works", {
+  val1 <- c("1", 2, 3)
+  val2 <- c('2', 4, '6', 8, '10')
+  expect_equal(c(1,2,3), .NumericApplyRange(val1, NULL))
+  expect_equal(c(1,2,3), .NumericApplyRange(val1, c(1,3)))
+  expect_equal(c(1,2,3), .NumericApplyRange(val1, c(1,3,5))) # spurious val 5
+  expect_equal(c(2,4,6,8,10), .NumericApplyRange(val2, NULL))
+  expect_equal(c(4,6,8), .NumericApplyRange(val2, c(2,4)))
+})
+
+test_that("apply_ranges_works", {
+  data <- matrix(c(1,2,3,
+                   4,5,6,
+                   7,8,9), nrow = 3, ncol = 3, byrow=TRUE)
+
+  # Various ways to apply ranges that change nothing.
+  expect_equal(data, .ApplyRanges(data, NULL, NULL))    # no ranges
+  expect_equal(data, .ApplyRanges(data, c(1,3), NULL))  # full row range
+  expect_equal(data, .ApplyRanges(data, NULL, c(1,3)))  # full col range
+  expect_equal(data, .ApplyRanges(data, c(1,3), c(1,3))) # both full ranges
+
+  # Trim just rows or just columns.
+  expect_equal(matrix(c(1,2,3,
+                        4,5,6), nrow=2, ncol=3, byrow=TRUE),
+               .ApplyRanges(data, c(1,2), NULL))
+  expect_equal(matrix(c(1,2,
+                        4,5,
+                        7,8), nrow=3, ncol=2, byrow=TRUE),
+               .ApplyRanges(data, NULL, c(1,2)))
+
+  # Trim both rows and columns.
+  expect_equal(matrix(c(1,2,
+                        4,5), nrow=2, ncol=2, byrow=TRUE),
+               .ApplyRanges(data, c(1,2), c(1,2)))
+})
+
+test_that("extended_scores_works", {
+  # Note that we expect the kriging in this simple case to preserve
+  # the corner values 1,2,3,4.  In between, we expect to see additional
+  # interpolated data points that show a progression from one original
+  # value to the next.
+  data <- matrix(c(1,2,
+                   3,4), nrow=2, ncol=2, byrow=TRUE)
+  print(c(data))
```

Comment:
not needed here?

Reply:
Quite right.  It was useful during development, but I meant to take it out before now.